plugins/portal-history-ornaments.js

```diff
@@ -0,0 +1,244 @@
+    .append('<div class="leaflet-control leaflet-bar" id="toggleHistoryButton"> '
+      +'<a onclick="window.plugin.portalHistoryOrnaments.toggleHistory(true); return false;" '
+      +'id="toggleHistory" class="'+isClass+'" title="History toggle"></a></div>');
+*/
+};
+
+function toggleHistory(keepUIbutton) {
+  var button = $('#toggleHistory');
+
+  portalsHistory.settings.drawMissing = !portalsHistory.settings.drawMissing
+  localStorage[KEY_SETTINGS] = JSON.stringify(portalsHistory.settings);
+  drawAllFlags();
+
+  if (button.hasClass('normHistory')) {
+    button.removeClass('normHistory');
+    button.addClass('revHistory');
+  } else {
+    button.addClass('normHistory');
+    button.removeClass('revHistory');
+  }
+};
+
+function svgToIcon (str, s) {
+  var url = ("data:image/svg+xml," + encodeURIComponent(str)).replace(/#/g, '%23');
+  return new L.Icon({
+    iconUrl: url,
+    iconSize: [s, s],
+    iconAnchor: [s / 2, s / 2],
+    className: 'no-pointer-events', //allows users to click on portal under the unique marker
+  })
+}
+
+function loadSettings() {
+  try {
+    portalsHistory.settings = JSON.parse(localStorage[KEY_SETTINGS]);
+  } catch (e) {
+    portalsHistory.settings = {
+      drawMissing: false,
+      showVisited: true,
+      showCaptured: true,
+      showScouted: false,
+    };
+  }
+}
+
+function toggleDisplayMode () {
+  dialog({
+    html: `<div id="portal-history-settings">
+<div>
+<select id="portal-history-settings--display-mode">
+  <option value="received" ${portalsHistory.settings.drawMissing?'':'selected'}>Show uniques received</option>
+  <option value="missing" ${portalsHistory.settings.drawMissing?'selected':''}>Show missing uniques</option>
+</select>
+</div>
+<div><label style="color:red;"><input type="checkbox" id="portal-history-settings--show-visited" 
+  ${portalsHistory.settings.showVisitedCaptured?'checked':''}> Show visited/captured</label></div>
+<div><label style="color:violet;"><input type="checkbox" id="portal-history-settings--show-scouted" 
+  ${portalsHistory.settings.showScouted?'checked':''}> Show Scout Controlled</label></div>
+</div>`,
+
+    title: 'Portal History Settings',
+    id: 'plugin-portal-history-flags',
+    width: 'auto',
+    closeCallback: function () {
+      var elMode = document.getElementById('portal-history-settings--display-mode');
+      var elVisitedCaptured = document.getElementById('portal-history-settings--show-visited');
+      var elScouted = document.getElementById('portal-history-settings--show-scouted');
+
+      portalsHistory.settings.drawMissing = elMode.value === 'missing';
+      portalsHistory.settings.showVisitedCaptured = elVisitedCaptured.checked;
+      portalsHistory.settings.showScouted = elScouted.checked;
+
+      localStorage[KEY_SETTINGS] = JSON.stringify(portalsHistory.settings);
+      portalsHistory.drawAllFlags();
+    }
+  });
+}
+  
+function createIcons () {
+//  var LEVEL_TO_RADIUS = [7, 7, 7, 7, 8, 8, 9, 10, 11];
```

Comment:
to indicate the original portal marker radius.
Comment should be removed before RTM

Reply:
done

Reply:
Why do you need that array? What if values are changed by some plugin?

Reply:
to adjust the base size of the ornaments.
if some plugin changes the portalMarker's radius this will not be effected.
The most common way to resize a portalMarker is a highlighter, but we have no chance to know the final size before we create the icons.

Reply:
Comment removed